GiniVision/Classes/Networking/SinglePageDocumentsService.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  SinglePageDocumentsService.swift
+//  GiniVision
+//
+//  Created by Enrique del Pozo Gómez on 3/29/18.
+//
+
+import Foundation
+import Gini_iOS_SDK
+
+final class SinglePageDocumentsService: DocumentServiceProtocol {
+
+    var compositeDocument: GINIDocument?
+    var giniSDK: GiniSDK
+    
+    var partialDocumentInfo: PartialDocumentInfo?
+    var pendingAnalysisHandler: AnalysisCompletion?
+    var analysisCancellationToken: BFCancellationTokenSource?
+    
+    init(sdk: GiniSDK) {
+        self.giniSDK = sdk
+    }
+    
+    func startAnalysis(completion: @escaping AnalysisCompletion) {
+        guard let partialDocumentInfo = partialDocumentInfo else {
+            pendingAnalysisHandler = completion
+            return
+        }
+
+        fetchExtractions(for: [partialDocumentInfo], completion: completion)
+    }
+    
+    func cancelAnalysis() {
+        compositeDocument = nil
+        partialDocumentInfo = nil
+        pendingAnalysisHandler = nil
+        
+        analysisCancellationToken?.cancel()
+        analysisCancellationToken = nil
+    }
+    
+    func remove(document: GiniVisionDocument) {
+        if let documentId = partialDocumentInfo?.documentId {
+            deletePartialDocument(withId: documentId)
+        }
+        cancelAnalysis()
+    }
+    
+    func update(parameters: [String: Any], for document: GiniVisionDocument) {
+        partialDocumentInfo?.updateAdditionalParameters(with: parameters)
+    }
+    
+    func upload(document: GiniVisionDocument,
+                completion: UploadDocumentCompletion?) {
+        partialDocumentInfo = PartialDocumentInfo()
+        
+        createDocument(from: document,
+                       fileName: "fileName") { result in
```

Comment:
File name is "fileName"...
I really don't like it - it seems to me that the file name property is useless.